providers/config.rb

```diff
@@ -17,7 +17,8 @@ def install_haproxy_config
     mode 00644
     variables(
       :defaults_options => defaults_options,
-      :defaults_timeouts => defaults_timeouts
+      :defaults_timeouts => defaults_timeouts,
+      :conf_template_variables => new_resource.conf_template_variables
```

Comment:
I do not see any corresponding updates to the default template to make use of this new variable. Is it expected to only be used by some other "wrapper cookbook" which would implement a custom template?

It seems if that is the case and the default is to do nothing because the default template (`haproxy.cfg.erb`) does not implement any defaults for the variable then it should be called out very specifically in the documentation. 

However, I think that it would be even better to actually implement an update to the default template to add the additional configuration if the attribute is not nil. 

Right now if someone sets that attribute without the knowledge that they need to have a custom template to override the default these changes would fail silently.